### PR TITLE
new presence: LoL Esports

### PR DESCRIPTION
**Pull request by mertkaraca34**
writes in Turkish because it was taken before. currently set to english.

image

 
image

### CODE REVIEW

File: websites/L/LoL Espor/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "$schema": "https://schemas.premid.app/metadata/1.0",
+  "author": {
+    "name": "mertkaraca.mp4",
+    "id": "749681979741569052"
+  },
+  "service": "LoL Esports",
```
Comment (Bas950) [inline websites/L/LoL Espor/dist/metadata.json:7]: update the folder name, and the name in discord, and upload new proof of it working then

File: websites/L/LoL Espor/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  "description": {
+    "tr":"League of legends oyununun Espor karşılaşmalarını canlı olarak izleyebileceğiniz platform.",
+    "en": "The platform where you can watch the Esports matches of the League of Legends game live."
+  },
+  "url": "lolesports.com",
+  "version": "1.0.0",
+  "logo": "https://i.imgur.com/CFlhiDo.jpg",
+  "thumbnail": "https://i.imgur.com/tmhf2Cm.jpg",
+  "color": "#3b1b8a",
+  "tags": ["lol", "leagueoflegends","espor","development","coding"],
```
Comment (Bas950) [inline websites/L/LoL Espor/dist/metadata.json:17]: espor > esports

File: websites/L/LoL Espor/presence.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    presenceData.details = "Schedule";
+    presenceData.state = "Viewing mathces.";
+    presenceData.startTimestamp = time;
+  }else if(path.startsWith("/live/") ){
+    const test = document.querySelector("div.teams").textContent;
+      presenceData.details = "Live!";
+    presenceData.state = test.replace("VS", " vs ") + " is watching.";
+    presenceData.startTimestamp = time;
+
+
```
Comment (ririxichan) [inline websites/L/LoL Espor/presence.ts:29]: fix: remove spaces and do better tab for 25 line

File: websites/L/LoL Espor/presence.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    presenceData.details = "News";
+    presenceData.state = "Looking at a news.";
+    presenceData.startTimestamp = time;
+  } else if (path.startsWith("/schedule")) {
+    presenceData.details = "Schedule";
+    presenceData.state = "Viewing mathces.";
+    presenceData.startTimestamp = time;
+  }else if(path.startsWith("/live/") ){
+    const test = document.querySelector("div.teams").textContent;
+      presenceData.details = "Live!";
```
Comment (Bas950) [inline websites/L/LoL Espor/presence.ts:25]: Give variables a better name then "test"

File: websites/L/LoL Espor/presence.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    presenceData.details = "Reading news";
+    presenceData.state = "News: " + document.querySelector("div.title").textContent;
+    presenceData.startTimestamp = time;
+  }else if(path.startsWith("/vods/")){
+    presenceData.details = "Repeats";
+    presenceData.state = "Looking match histories."
+    presenceData.startTimestamp = time;
+  }else if(path.startsWith("/vod/")){
+    let test = document.querySelector("div.teams").textContent;
+    presenceData.details = "Watching repeats";
```
Comment (Bas950) [inline websites/L/LoL Espor/presence.ts:40]: const, and give variable a better name

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Review by ririxichan — CHANGES_REQUESTED** [review]

File: websites/L/LoL Espor/presence.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    presenceData.details = "News";
+    presenceData.state = "Looking at a news.";
+    presenceData.startTimestamp = time;
+  } else if (path.startsWith("/schedule")) {
+    presenceData.details = "Schedule";
+    presenceData.state = "Viewing mathces.";
+    presenceData.startTimestamp = time;
+  }else if(path.startsWith("/live/") ){
+    let tt = document.querySelector("div.teams").textContent;
+    presenceData.details = "Live!";
```
Comment (Bas950) [inline websites/L/LoL Espor/presence.ts:24]: tt? Do something like, "teams"

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/L/LoL Espor/presence.ts (hunk 5 above)
Comment (mertkaraca34) [reply websites/L/LoL Espor/presence.ts:37]: done

File: websites/L/LoL Espor/presence.ts [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+  }else if(path.startsWith("/vods/")){
+    presenceData.details = "Repeats";
+    presenceData.state = "Looking match histories."
+    presenceData.startTimestamp = time;
+  }else if(path.startsWith("/vod/")){
+    let test = document.querySelector("div.teams").textContent;
+    presenceData.details = "Watching repeats";
+    presenceData.state = test.replace("VS", " vs ") + " watching the match replay." 
+    presenceData.startTimestamp = time;
+  }else if(path.startsWith("/standings/")){
```
[the remark this answers is not in the document]
Comment (mertkaraca34) [reply websites/L/LoL Espor/presence.ts]: done

File: websites/L/LoL Espor/presence.ts (hunk 6 above)
Comment (mertkaraca34) [reply websites/L/LoL Espor/presence.ts]: done

File: websites/L/LoL Espor/presence.ts (hunk 4 above)
Comment (mertkaraca34) [reply websites/L/LoL Espor/presence.ts]: done

File: websites/L/LoL Espor/dist/metadata.json (hunk 2 above)
Comment (mertkaraca34) [reply websites/L/LoL Espor/dist/metadata.json]: done

File: websites/L/LoL Espor/dist/metadata.json (hunk 1 above)
Comment (mertkaraca34) [reply websites/L/LoL Espor/dist/metadata.json:7]: done

File: websites/L/LoL Espor/presence.ts [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+  } else if (path === "/news") {
+    presenceData.details = "News";
+    presenceData.state = "Looking at a news.";
+    presenceData.startTimestamp = time;
+  } else if (path.startsWith("/schedule")) {
+    presenceData.details = "Schedule";
+    presenceData.state = "Viewing mathces.";
+    presenceData.startTimestamp = time;
+  }else if(path.startsWith("/live/") ){
+    let liveqs = document.querySelector("div.teams").textContent;
```
Comment (ririxichan) [inline websites/L/LoL Espor/presence.ts:23]: let -> const

**Review by ririxichan — CHANGES_REQUESTED** [review]

File: websites/L/LoL Espor/presence.ts [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    presenceData.details = "Home";
+    presenceData.state = "Nothing is done.";
+    presenceData.startTimestamp = time;
+  } else if (path === "/news") {
+    presenceData.details = "News";
+    presenceData.state = "Looking at a news.";
+    presenceData.startTimestamp = time;
+  } else if (path.startsWith("/schedule")) {
+    presenceData.details = "Schedule";
+    presenceData.state = "Viewing mathces.";
```
Comment (ririxichan) [inline websites/L/LoL Espor/presence.ts:20]: fix typo. matches.

**Review by ririxichan — CHANGES_REQUESTED** [review]

**Comment by ririxichan** [comment]
also change folder name to LoL Esports

File: websites/L/LoL Espor/presence.ts (hunk 9 above)
Comment (mertkaraca34) [reply websites/L/LoL Espor/presence.ts]: done

File: websites/L/LoL Espor/presence.ts (hunk 8 above)
Comment (mertkaraca34) [reply websites/L/LoL Espor/presence.ts]: done

File: websites/L/LoL Espor/presence.ts [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+    presenceData.details = "Live!";
+    presenceData.state = liveqs.replace("VS", " vs ") + " is watching.";
+    presenceData.startTimestamp = time;
+  }else if(path.startsWith("/article/")){
+    presenceData.details = "Reading news";
+    presenceData.state = "News: " + document.querySelector("div.title").textContent;
+    presenceData.startTimestamp = time;
+  }else if(path.startsWith("/vods/")){
+    presenceData.details = "Repeats";
+    presenceData.state = "Looking match histories.";
```
Comment (Bas950) [inline websites/L/LoL Espor/presence.ts:33]: Looking at match histories

File: websites/L/LoL Espor/presence.ts [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+    presenceData.state = "Viewing match.";
+    presenceData.startTimestamp = time;
+  }else if(path.startsWith("/live/") ){
+    const liveqs = document.querySelector("div.teams").textContent;
+    presenceData.details = "Live!";
+    presenceData.state = liveqs.replace("VS", " vs ") + " is watching.";
+    presenceData.startTimestamp = time;
+  }else if(path.startsWith("/article/")){
+    presenceData.details = "Reading news";
+    presenceData.state = "News: " + document.querySelector("div.title").textContent;
```
Comment (Bas950) [inline websites/L/LoL Espor/presence.ts:29]: Reading news article:
secondline just element textcontent

File: websites/L/LoL Espor/presence.ts [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+    presenceData.details = "Home";
+    presenceData.state = "Nothing is done.";
+    presenceData.startTimestamp = time;
+  } else if (path === "/news") {
+    presenceData.details = "News";
+    presenceData.state = "Looking at a news.";
+    presenceData.startTimestamp = time;
+  } else if (path.startsWith("/schedule")) {
+    presenceData.details = "Schedule";
+    presenceData.state = "Viewing match.";
```
Comment (Bas950) [inline websites/L/LoL Espor/presence.ts:20]: Viewing matches.

File: websites/L/LoL Espor/presence.ts [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+}),
+time = Math.floor(Date.now() / 1000),
+path = window.location.pathname;
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "logo"
+  };
+  if (path === "/") {
+    presenceData.details = "Home";
+    presenceData.state = "Nothing is done.";
```
Comment (Bas950) [inline websites/L/LoL Espor/presence.ts:12]: Viewing the homepage.
No state

File: websites/L/LoL Espor/presence.ts [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+    presenceData.state = "Looking match histories.";
+    presenceData.startTimestamp = time;
+  }else if(path.startsWith("/vod/")){
+    const teams = document.querySelector("div.teams").textContent;
+    presenceData.details = "Watching repeats";
+    presenceData.state = teams.replace("VS", " vs ") + " watching the match replay.";
+    presenceData.startTimestamp = time;
+  }else if(path.startsWith("/standings/")){
+    presenceData.details = "Standings";
+    presenceData.state = "Looking at standings.";
```
Comment (Bas950) [inline websites/L/LoL Espor/presence.ts:42]: Looking at the standings

**Review by Bas950 — CHANGES_REQUESTED** [review]
update the folder name, and the name in discord, and upload new proof of it working then

**Review by Bas950 — CHANGES_REQUESTED** [review]
Delete logs file and make the changes requested before.

File: websites/L/LoL Espor/presence.ts (hunk 10 above)
Comment (mertkaraca34) [reply websites/L/LoL Espor/presence.ts]: done

File: websites/L/LoL Espor/presence.ts (hunk 11 above)
Comment (mertkaraca34) [reply websites/L/LoL Espor/presence.ts]: done

File: websites/L/LoL Espor/presence.ts (hunk 12 above)
Comment (mertkaraca34) [reply websites/L/LoL Espor/presence.ts]: done

File: websites/L/LoL Espor/presence.ts (hunk 13 above)
Comment (mertkaraca34) [reply websites/L/LoL Espor/presence.ts]: done

File: websites/L/LoL Espor/presence.ts (hunk 14 above)
Comment (mertkaraca34) [reply websites/L/LoL Espor/presence.ts]: done

**Comment by mertkaraca34** [comment]
all issues fixed

File: logs [hunk 15]
```diff
@@ -0,0 +1,468 @@
...
+Author: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
+Date:   Mon Sep 28 06:14:59 2020 +0000
+
+    🔼 Bump @types/babel__core from 7.1.9 to 7.1.10
+    
+    Bumps [@types/babel__core](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/babel__core) from 7.1.9 to 7.1.10.
+    - [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
+    - [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/babel__core)
+    
+    Signed-off-by: dependabot[bot] <support@github.com>
```
Comment (Bas950) [inline logs:468]: Delete this file

**Review by Bas950 — CHANGES_REQUESTED** [review]
Change the name in discord, and upload new proof of it working then

**Comment by Bas950** [comment]
Hasn't made the requested changes in over a week

**Pull request closed without merging**